x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/common/EmbeddingRequestChunker.java

```diff
@@ -205,34 +281,63 @@ private void sendResponse() {
                 if (errors.get(i) != null) {
                     response.add(errors.get(i));
                 } else {
-                    response.add(merge(chunkedInputs.get(i), results.get(i)));
+                    response.add(mergeResults(i));
                 }
             }
 
             finalListener.onResponse(response);
         }
+    }
 
-        private InferenceChunkedTextEmbeddingFloatResults merge(
-            List<String> chunks,
-            AtomicArray<List<InferenceTextEmbeddingFloatResults.InferenceFloatEmbedding>> debatchedResults
-        ) {
-            var all = new ArrayList<InferenceTextEmbeddingFloatResults.InferenceFloatEmbedding>();
-            for (int i = 0; i < debatchedResults.length(); i++) {
-                var subBatch = debatchedResults.get(i);
-                all.addAll(subBatch);
-            }
+    private ChunkedInferenceServiceResults mergeResults(int resultIndex) {
+        return switch (embeddingType) {
+            case FLOAT -> mergeFloatResults(chunkedInputs.get(resultIndex), floatResults.get(resultIndex));
+            case BYTE -> mergeByteResults(chunkedInputs.get(resultIndex), byteResults.get(resultIndex));
+        };
+    }
```

Comment:
The thing thats happening here is that we are matching the inputs with the results, right?

Maybe we can rename this function to `mergeResultsWithInputs`?

Reply:
done